src/components/blocks/external-references/A.tsx

```diff
@@ -6,20 +6,32 @@ import { DOCUMENTATION_NAME } from '../../../../data/transform/constants';
 import { HtmlComponentProps } from '../../html-component-props';
 
 import './styles.css';
+import Img from './Img';
 
 const StyledGatsbyLink = ({ to, children, ...props }: Omit<GatsbyLinkProps<Record<string, unknown>>, 'ref'>) => (
-  <Link className="docs-link" to={to} {...props}>
+  <Link className="docs-link" data-testid="gatsby-link" to={to} {...props}>
     {children}
   </Link>
 );
 
 const A = ({ data, attribs }: HtmlComponentProps<'a'>): ReactElement => {
   const rawHref = attribs?.href;
+
+  // If there is an image inside the link with src same as href, then nuke <a> and render <img> only
+  if (Array.isArray(data)) {
+    const imageElement = data.find((item) => item.name === 'img');
+    // @ts-ignore - says src doesn't exist, but it does
```

Comment:
I would prefer to just have a guard here if possible - src might not always exist?